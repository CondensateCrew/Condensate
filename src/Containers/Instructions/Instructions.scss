@import "../../css/variables.scss";
.instructions {
  height: 100%;
  section {
    height: 80%; width: 80%;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: auto;
    p {
      font-size: $logo-text;
      animation: appearing-toggle 4.01s cubic-bezier(0.250, 0.460, 0.450, 0.940) infinite;
      opacity: 1;
      line-height: 1.4em;
    }
  }
}

@keyframes appearing-toggle {
  0% {
    opacity: 0;
  }
  40% {
    opacity: 1;
  }
  90% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
