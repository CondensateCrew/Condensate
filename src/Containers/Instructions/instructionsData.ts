interface IInstructionsData {
  instructions: string[],
  link: string
}

interface ITypesData {
  [key: string]: IInstructionsData
}

export const firstIinstructions: IInstructionsData  = {
  instructions: [
    'In the next 90 seconds choose 9 words',
    'Choose whichever words compel you',
    'They should elicit positive feelings',
    'Relax',
    'Enjoy'
  ],
  link: '/round-one'
};

export const secondIinstructions: IInstructionsData  = {
  instructions: [
    'You will have 90 seconds',
    'To write as many questions as you can',
    'Try to write the first thing that comes to your mind',
    'The more spontaneous and provocative the better!'
  ],
  link: '/round-two'
};

const types: ITypesData = {
  'first-rd': firstIinstructions,
  'second-rd': secondIinstructions
}

export default types
