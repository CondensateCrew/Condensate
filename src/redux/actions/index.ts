import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
