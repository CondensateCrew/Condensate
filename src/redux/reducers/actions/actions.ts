import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
