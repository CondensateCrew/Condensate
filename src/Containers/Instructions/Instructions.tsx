import './Instructions.scss';
import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { AppStore } from 'interfaces';
import Header from 'Components/Header/Header';
import { endInstructions, reverseInstructions, reverseTime } from 'redux/actions';
import { Redirect } from 'react-router-dom';
import types from './instructionsData';

interface Props {
  type: string
};

const Instructions:React.FC<Props> = ({ type }) => {
  const instructions: string[] = types[type].instructions;
  const link: string = types[type].link;
  const dispatch = useDispatch();
  const instructionsEnded: boolean = useSelector((store: AppStore) => store.instructionsEnded);
  const [ line, setLine ] = useState<string>('');
  const [ count, setCount ] = useState<number>(0);

  useEffect(() => {
    dispatch(reverseInstructions());
    dispatch(reverseTime());
    setLine(instructions[0]);
    setCount(1);
  }, [dispatch, instructions])

  useEffect(() => {
    const timer = setInterval(() => {
      setLine(instructions[count]);
      setCount(count + 1);
    }, 4000);

    return () => clearInterval(timer);
  }, [instructions, count]);

  useEffect(() => {
    if (count > instructions.length) dispatch(endInstructions());
  }, [ count, dispatch, instructions.length ]);

  return (
    <main className="instructions">
      <Header />
      <section>
        { instructionsEnded
          ? <Redirect to={link} />
          : <p>{line}</p>
        }
      </section>
    </main>
  );
}

export default Instructions;
