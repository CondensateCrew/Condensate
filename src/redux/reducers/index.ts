import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
