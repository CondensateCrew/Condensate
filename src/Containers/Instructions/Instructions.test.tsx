import React from 'react';
import Instructions from './Instructions';
import { mount } from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import { endInstructions, reverseInstructions, reverseTime } from 'redux/actions';
import { act } from '@testing-library/react';

const mockDispatch = jest.fn();
let mockValue: boolean;

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => mockValue
}));

jest.useFakeTimers();

jest.mock('redux/actions');

describe('Instructions component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  
  it('should match the snapshot for type first-rd', () => {
    mockValue = false;
    const wrapper = mount(
      <Instructions type="first-rd" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if instructions ended for type first-rd', () => {
    mockValue = true;
    const wrapper = mount(
      <MemoryRouter keyLength={0} initialEntries={[{
        pathname: '/', key: 'testKey'
      }]}>
        <Instructions type="first-rd" />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for type second-rd', () => {
    mockValue = false;
    const wrapper = mount(
      <Instructions type="second-rd" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if instructions ended for type second-rd', () => {
    mockValue = true;
    const wrapper = mount(
      <MemoryRouter keyLength={0} initialEntries={[{
        pathname: '/', key: 'testKey'
      }]} >
        <Instructions type="second-rd" />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe("function calls", () => {
    let wrapper: any;

    beforeEach(() => {
      mockValue = false;
      wrapper = mount(
        <Instructions type="second-rd" />
      );
    });

    it('should call reverseInstructions after rendering', () => {
      expect(reverseInstructions).toHaveBeenCalled();
    });

    it('should call reverseTime after rendering', () => {
      expect(reverseTime).toHaveBeenCalled();
    });

    it('should call endInstructions after rendering', () => {
      act(() => {
        jest.advanceTimersByTime(30000);
      });

      expect(endInstructions).toHaveBeenCalled();
    });
  });
});
