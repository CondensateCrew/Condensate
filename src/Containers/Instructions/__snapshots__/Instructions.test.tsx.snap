// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions component should match the snapshot for type first-rd 1`] = `
<Instructions
  type="first-rd"
>
  <main
    className="instructions"
  >
    <Header>
      <header
        className="round-header"
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="back"
            src="down.svg"
          />
          to dashboard
        </button>
        <h1>
          Condensate
        </h1>
      </header>
    </Header>
    <section>
      <p>
        In the next 60 seconds choose 9 words
      </p>
    </section>
  </main>
</Instructions>
`;

exports[`Instructions component should match the snapshot for type second-rd 1`] = `
<Instructions
  type="second-rd"
>
  <main
    className="instructions"
  >
    <Header>
      <header
        className="round-header"
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="back"
            src="down.svg"
          />
          to dashboard
        </button>
        <h1>
          Condensate
        </h1>
      </header>
    </Header>
    <section>
      <p>
        You will have 60 seconds
      </p>
    </section>
  </main>
</Instructions>
`;

exports[`Instructions component should match the snapshot if instructions ended for type first-rd 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/round-one",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/round-one",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Instructions
      type="first-rd"
    >
      <main
        className="instructions"
      >
        <Header>
          <header
            className="round-header"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="back"
                src="down.svg"
              />
              to dashboard
            </button>
            <h1>
              Condensate
            </h1>
          </header>
        </Header>
        <section>
          <Redirect
            to="/round-one"
          >
            <Lifecycle
              onMount={[Function]}
              onUpdate={[Function]}
              to="/round-one"
            />
          </Redirect>
        </section>
      </main>
    </Instructions>
  </Router>
</MemoryRouter>
`;

exports[`Instructions component should match the snapshot if instructions ended for type second-rd 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/round-two",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/round-two",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Instructions
      type="second-rd"
    >
      <main
        className="instructions"
      >
        <Header>
          <header
            className="round-header"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="back"
                src="down.svg"
              />
              to dashboard
            </button>
            <h1>
              Condensate
            </h1>
          </header>
        </Header>
        <section>
          <Redirect
            to="/round-two"
          >
            <Lifecycle
              onMount={[Function]}
              onUpdate={[Function]}
              to="/round-two"
            />
          </Redirect>
        </section>
      </main>
    </Instructions>
  </Router>
</MemoryRouter>
`;
